controllers/UserController.js

```diff
@@ -0,0 +1,83 @@
+import User from '../models/User';
+import passport from 'passport/lib';
+
+export default class UserController {
+
+    static getUser(req, res, next) {
+        User.findById(req.payload.id)
+            .then(user => {
+                if (!user) {
+                    return res.sendStatus(401);
+                }
+                return res.json({user: UserController.toAuthJSON(user.dataValues)});
+            })
+            .catch(next);
+    }
+
+    static updateUser(req, res, next) {
+        User.findById(req.payload.id)
+            .then(user => {
+                if (!user) {
+                    return res.sendStatus(401);
+                }
+
+                // only update fields that were actually passed...
+                if (typeof req.body.user.username !== 'undefined') {
```

Comment:
I think u can use `if (req.body.user.username)` instead of `if (typeof req.body.user.username !== 'undefined')`